recent_topics: Presence status and emoji for PMs in recent topics.

Riken-Shah opened the pull request:
<!-- Describe your pull request here.-->
Fixes https://github.com/zulip/zulip/issues/23262



<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
<img width="872" alt="Screenshot 2022-10-27 at 2 25 22 PM" src="https://user-images.githubusercontent.com/63820270/198246238-d6ca7ae3-cc05-4d7e-bf00-08577efcea7e.png">
<img width="872" alt="Screenshot 2022-10-27 at 2 25 27 PM" src="https://user-images.githubusercontent.com/63820270/198246243-21c42604-aa70-4521-b76d-cdafb7bd501b.png">
<img width="872" alt="Screenshot 2022-10-27 at 2 25 32 PM" src="https://user-images.githubusercontent.com/63820270/198246249-e6a0010d-51e6-44f6-8f85-d27700e369fd.png">
<img width="872" alt="Screenshot 2022-10-27 at 2 25 34 PM" src="https://user-images.githubusercontent.com/63820270/198246253-63f906ec-10ce-4bae-82ba-4fb4679b3339.png">



**Self-review checklist**


<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

Riken-Shah commented:
@alya ready for the initial review.

timabbott commented on static/styles/zulip.css:
```diff
@@ -124,6 +124,7 @@ pre {
 
     .tippy-content {
         font-size: 13px;
+        font-weight: 400;
```
Can you explain this change? It seems like it'd apply to _all_ tooltips in the app.

Riken-Shah replied:
Without it the tooltip content had the bold effect, 
I have fixed it by moving it to `recent_topics.css`.

alya commented:
Thanks!

It looks like this PR starts showing your own name in list of conversations, which needs to be fixed.

alya commented:
I also noticed that we're treating bots differently from how the are treated in the left sidebar. Let's just make Recent topics match the left sidebar for how bots are handled, which seems to be always showing a green presence dot, and listing the owner instead of the status.

![Screen Shot 2022-10-20 at 11 47 57 AM](https://user-images.githubusercontent.com/2090066/197032561-c34a688f-c1b6-41d5-bfae-b53a0cfab704.png)

timabbott commented:
@Riken-Shah do you think you'll have time to fix those bugs in the next couple days? If not we should probably get another contributor involved to help.

Riken-Shah commented:
Hey @timabbott, 
I'll work on it! Just busy with something last few days, will pick it up shortly.

alya commented:
Hm, I'm still seeing the user's own name in some PM conversations, though not others now. I don't know what the pattern is, but definitely something we need to fix (see bottom line of screenshot). It should only show your own name under Topic for 1:1 conversations with yourself (see 3rd line, which is correct).

![Screen Shot 2022-10-28 at 4 56 14 PM](https://user-images.githubusercontent.com/2090066/198751907-3e92eb89-5817-4014-9714-cdc04dff57b4.png)

alya commented:
For the presence tooltip, let's make a whole rectangle around the presence dot be the hover target. Otherwise it's too hard to trigger it.

Riken-Shah commented:
Sorry for the late response!

> Hm, I'm still seeing the user's own name in some PM conversations, though not others now. I don't know what the pattern is,

Yeah, my bad I was excluding last message sender id instead of the logged in user. Fixed it!

> For the presence tooltip, let's make a whole rectangle around the presence dot be the hover target. Otherwise it's too hard to trigger it.

Done!

alya commented:
Thanks! A few more comments:

1. I noticed that we need to drop the tooltip for group PMs:

![Screen Shot 2022-11-03 at 12 21 45 PM](https://user-images.githubusercontent.com/2090066/199815067-7bd86528-4a96-415e-999a-c9e53de9d668.png)


2. The hover target placement looks to be off; it should be centered around the presence dot.

![Screen Shot 2022-11-03 at 12 24 25 PM](https://user-images.githubusercontent.com/2090066/199816034-dd7391c0-6193-4902-976a-2f8eccf5facb.png)

3. I stepped away from my computer with the dev environment open to this PR, and when I came back, it was showing an error:

![Screen Shot 2022-11-03 at 12 26 49 PM](https://user-images.githubusercontent.com/2090066/199816235-25deedba-0199-4168-89bb-5b6ead096833.png)

Riken-Shah commented:
> 1.I noticed that we need to drop the tooltip for group PMs:

Fixed

> The hover target placement looks to be off; it should be centered around the presence dot.

<img width="872" alt="Screenshot 2022-11-05 at 2 11 05 PM" src="https://user-images.githubusercontent.com/63820270/200111554-c05ed906-37d5-4b55-bb57-f071929e78bf.png">

> I stepped away from my computer for a minute with the dev environment open to this PR, and when I came back, it was showing an error:

I suspect it happend because user string ids were empty, I added a condition to handle this. It shouldn't happen again mostly, but you can give it a try.

alya commented:
Hm, looking at the updated hover target, could you re-add the extra space above/below, like you had before? The new one looks to be tight to the dot in the vertical dimension.

timabbott commented:
What we're looking for is a hover area that is a square around the circle, with the square click target region and visible circle sharing the same center.

amanagr commented:
@Riken-Shah I will try to address any further reviews of this PR! Thanks for your work on this till now!

amanagr commented on static/styles/recent_topics.css:
```diff
@@ -70,13 +70,6 @@
             padding-right: 3px;
```
<img width="440" alt="image" src="https://user-images.githubusercontent.com/25124304/200843023-b3b9fd2c-69c4-4e8c-8db4-1c96d01320ce.png">

amanagr commented:
Should be better now.
<img width="124" alt="image" src="https://user-images.githubusercontent.com/25124304/200843115-646c41e1-f7ac-42fe-9576-cfbcb7083656.png">

alya commented:
Cool, the hover area feels pretty good!

alya commented:
However, I still (again?) incorrectly see a tooltip on group PM icons:

<img width="508" alt="Screen Shot 2022-11-09 at 2 57 18 PM" src="https://user-images.githubusercontent.com/2090066/200960126-cf022aa2-7e47-47f1-b9f9-50a018e3b2ad.png">

timabbott commented:
I agree the tooltip being above is probably best. It's reusing left sidebar code for the rendering which is this; we probably want to just pass `placement` as a parameter for this case.

```
        let placement = "left";
        let observer;
        if (window.innerWidth < media_breakpoints_num.md) {
            // On small devices display tooltips based on available space.
            // This will default to "bottom" placement for this tooltip.
            placement = "auto";
        }
```

amanagr commented:
Resolved both the issues. Thanks!
